server/region_handler.go

```diff
@@ -90,11 +90,13 @@ const (
 func (s *Server) newRegionHandler() (hanler *regionHandler, err error) {
 	var tikvStore kvStore
 	store, ok := s.driver.(*TiDBDriver)
-	if ok {
-		tikvStore, ok = store.store.(kvStore)
-	}
 	if !ok {
-		err = errors.New("Invalid KvStore")
+		log.Fatal("Invalid KvStore with illegal driver")
+		return
+	}
+
+	if tikvStore, ok = store.store.(kvStore); !ok {
+		log.Fatal("Invalid KvStore with illegal store")
```

Comment:
If so, why not just `panic`?
There is little reason to use `log.Fatal`. It call `os.Exit` behind, which will exit immediately **without** calling deferred functions, which may lead to resource leaking or something. Also, it doesn't print out the calling stack.